modules/acquia_cms_search/acquia_cms_search.module

```diff
@@ -116,6 +138,61 @@ function acquia_cms_search_entity_insert() {
   }
 }
 
+/**
+ * Import views for content types.
+ */
+function _acquia_cms_search_import_content_type_views(NodeTypeInterface $node_type){
+  $content_views = [
+    'article' => 'articles',
+    'event' => 'events',
+    'person' => 'people',
+    'place' => 'places',
+  ];
+  if(!in_array($node_type->id(), array_keys($content_views))) {
+    return;
+  }
+  // Load the view.
+  $view = View::load($content_views[$node_type->id()]);
+  if (!$view) {
+    // import the view.
+    $source = new FileStorage(__DIR__ . '/config/optional');
+    $entity_storage = \Drupal::entityTypeManager()->getStorage('view');
+    $view_config = $source->read('views.view.' . $content_views[$node_type->id()]);
+    // Validate dependencies exist.
+    if (!empty($view_config['dependencies'])) {
+      $module_handler = \Drupal::service('module_handler');
+      foreach ($view_config['dependencies']['module'] as $dependency) {
+        if (!$module_handler->moduleExists($dependency)) {
+          return;
+        }
+      }
+      foreach ($view_config['dependencies']['config'] as $dependency) {
+        if (!configExists($dependency)) {
+          return;
+        }
+      }
+    }
+
+    // Create the view entity.
+    $entity = $entity_storage->createFromStorageRecord($view_config);
+    $entity->save();
+  }
+}
+
+/**
+ * Check if a configuration exists.
+ *
+ * @param string $config_name
+ *   The name of the configuration to check.
+ *
+ * @return bool
+ *   TRUE if the configuration exists, FALSE otherwise.
+ */
+function configExists($config_name) {
```

Comment:
this method is called only at one instance.. can we directly implement the code block where this method called.
Pros: We can avoid the introducing of new method.

Just a suggestion!!